### PR TITLE
Fail tests with an actionable message when FLUTTER_ROOT is not set

### PR DESCRIPTION
**Pull request by goderbauer**
Also: Fix some analyzer warnings.

### CODE REVIEW

**Comment by cbracken** [comment]
lgtm

File: packages/flutter_tools/lib/src/asset.dart [hunk 1]
```diff
@@ -336,7 +336,7 @@ Map<_Asset, List<_Asset>> _parseAssets(
   if (manifestDescriptor == null)
     return result;
 
-  excludeDirs = excludeDirs.map(
+  excludeDirs = excludeDirs.map<String>(
     (String exclude) => fs.path.absolute(exclude) + fs.path.separator).toList();
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/asset.dart]: nit: this line is wrapped weird. It's not clear where the function arguments end. Maybe put a newline before the penultimate ) ?

File: packages/flutter_tools/test/forbidden_imports_test.dart [hunk 2]
```diff
@@ -8,12 +8,17 @@ import 'package:test/test.dart';
 
 void main() {
   setUp(() {
-    String flutterTools = fs.path.join(platform.environment['FLUTTER_ROOT'],
-        'packages', 'flutter_tools');
+    String flutterRoot = platform.environment['FLUTTER_ROOT'];
+    if (flutterRoot == null) {
```
Comment (Hixie) [inline packages/flutter_tools/test/forbidden_imports_test.dart]: nit: no braces needed here

File: packages/flutter_tools/test/forbidden_imports_test.dart [hunk 3]
```diff
@@ -8,12 +8,17 @@ import 'package:test/test.dart';
...
+      throw new Exception('Please set FLUTTER_ROOT env var before running tests.');
+    }
```
Comment (cbracken) [inline packages/flutter_tools/test/forbidden_imports_test.dart]: nit: no curly brackets around single-line ifs.

**Pull request merged**